user-manager/src/main/java/com/qpay/usermanager/service/impl/CustomerServiceImpl.java

```diff
@@ -9,51 +13,78 @@
 import com.qpay.usermanager.service.exception.CustomerAlreadyExistsException;
 import com.qpay.usermanager.service.exception.CustomerNotFoundException;
 import com.qpay.usermanager.service.exception.EmailAlreadyExistsException;
+import io.micrometer.common.util.StringUtils;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.dao.DataIntegrityViolationException;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
 
 @Service
 @Slf4j
 @RequiredArgsConstructor
+@Transactional(isolation = Isolation.REPEATABLE_READ)
```

Comment:
Why repeatable read and not read commit ?

Reply:
i thought repeatable would be better, if parallel requests saved duplicate email, but i tested with REPEATABLE_READ and it works correctly